tests/unit/amo/components/TestConfirmButton.js

```diff
@@ -1,170 +1,154 @@
+    expect(screen.getByClassName('ConfirmationDialog')).toBeInTheDocument();
   });
 
   it('configures ConfirmationDialog', () => {
     const cancelButtonText = 'Nevermind, take me back';
-    const cancelButtonType = 'alert';
     const confirmButtonText = 'Do it!';
-    const confirmButtonType = 'alert';
     const message = 'Do you really want to cancel?';
-    const puffyButtons = true;
 
-    const root = renderWithDialog({
+    renderWithDialog({
       cancelButtonText,
-      cancelButtonType,
+      cancelButtonType: 'alert',
       confirmButtonText,
-      confirmButtonType,
+      confirmButtonType: 'alert',
       message,
-      puffyButtons,
+      puffyButtons: true,
     });
 
-    const confirmDialog = root.find(ConfirmationDialog);
-    expect(confirmDialog).toHaveProp('cancelButtonText', cancelButtonText);
-    expect(confirmDialog).toHaveProp('cancelButtonType', cancelButtonType);
-    expect(confirmDialog).toHaveProp('confirmButtonText', confirmButtonText);
-    expect(confirmDialog).toHaveProp('confirmButtonType', confirmButtonType);
-    expect(confirmDialog).toHaveProp('message', message);
-    expect(confirmDialog).toHaveProp('puffyButtons', puffyButtons);
+    expect(screen.getByText(message)).toBeInTheDocument();
+
+    const dialogCancelButton = screen.getByRole('button', {
+      name: cancelButtonText,
+    });
+    expect(dialogCancelButton).toHaveClass('Button--alert');
+    expect(dialogCancelButton).toHaveClass('Button--puffy');
+
+    const dialogConfirmButton = screen.getByRole('button', {
+      name: confirmButtonText,
+    });
+    expect(dialogConfirmButton).toHaveClass('Button--alert');
+    expect(dialogConfirmButton).toHaveClass('Button--puffy');
   });
 
   it('hides the default button after it is clicked', () => {
-    const { store } = dispatchClientMetadata();
-    const root = render({ store });
+    render();
 
-    root
-      .find('.ConfirmButton-default-button')
-      .simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
-
-    expect(root.find('.ConfirmButton-default-button')).toHaveLength(0);
+    const button = screen.getByRole('button', {
+      name: defaultChildText,
+    });
+    userEvent.click(button);
+    expect(button).not.toBeInTheDocument();
   });
 
-  it('hides ConfirmationDialog via onCancel', () => {
-    const { store } = dispatchClientMetadata();
-    const root = renderWithDialog({ store });
+  it('hides ConfirmationDialog on cancel', () => {
+    renderWithDialog();
 
-    const onCancel = getDialogProp(root, 'onCancel');
-    // Simulate a cancellation.
-    onCancel(createFakeEvent());
+    userEvent.click(screen.getByRole('button', { name: 'Cancel' }));
 
-    applyUIStateChanges({ root, store });
-    expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
-    expect(root.find(ConfirmationDialog)).toHaveLength(0);
+    expect(
+      screen.queryByClassName('ConfirmationDialog'),
```

Comment:
Is there something else we can assert about in place of this class name? E.g., the warning message or a button? Ideally we would first assert it's present, and then that it is gone after clicking the button.